ansible_rulebook/rule_set_runner.py

```diff
@@ -309,14 +309,14 @@ async def _call_action(
         rule_uuid: str,
         rule_run_at: str,
         action: str,
-        action_args: Dict,
+        immutable_action_args: Dict,
         variables: Dict,
         inventory: str,
         hosts: List,
         rules_engine_result,
     ) -> None:
-
         logger.info("call_action %s", action)
+        action_args = immutable_action_args.copy()
```

Comment:
Naming it "immutable" can be confusing because it suggests must be immutable but there is no any validation, the value is copied later and the type hint stills be "dict". 
Also `copy` method doesn't make it immutable. And it's the copied value which is supposed to be immutable, not the received argument. 

If the immutability constraint will be addressed inside the function, I suggest to keep the original name of the param and enforce the immutability with MappingProxyType adding the "immutable" prefix to the variable. 
```
from types import MappingProxyType
immutable_action_args = MappingProxyType(action_args.copy())
```